.github/workflows/ci.yml

```diff
@@ -0,0 +1,77 @@
+name: CI
+
+on:
+  schedule:
+    - cron:  '0 5 30 * *'
+  pull_request:
+    paths-ignore:
+    - 'docs/**'
+    - 'charts/**'
+    - 'scripts/**'
+    - '*.md'
+
+env:
+  GOARCH: amd64
+  CGO_ENABLED: 0
+  SETUP_GO_VERSION: '^1.18'
+
+jobs:
+  unit-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      -
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      -
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.SETUP_GO_VERSION }}
+      -
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      -
+        name: unit-test
+        run: go test -shuffle=on $(go list ./... | grep -v /e2e | grep -v /integration)
+
+  integration-test:
+    needs: [unit-test]
+    runs-on: ubuntu-latest
+
+    steps:
+      -
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      -
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          repository: rancher/fleet-examples
+          path: fleet-examples
+      -
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.SETUP_GO_VERSION }}
+      -
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      -
+        name: Install Ginkgo CLI
+        run: go install github.com/onsi/ginkgo/v2/ginkgo@v2.1
+      -
+        name: Test
+        run: ginkgo integration
```

Comment:
Since we don't need a live cluster to test `fleet apply`, I think this should just be a `./scripts/integration` script that is defined similarly to `./scripts/test` and called in `./scripts/ci`:

https://github.com/rancher/fleet/blob/976fd120fb15eb0d310540af436b52e1f5817821/scripts/ci#L7

Installing Ginkgo CLI can be done in the [`Dockerfile.dapper`](https://github.com/rancher/fleet/blob/976fd120fb15eb0d310540af436b52e1f5817821/Dockerfile.dapper) itself and we can avoid creating another GH workflow for this since it will just be part of running `ci` itself.